src/Metrics.ts

```diff
@@ -307,6 +307,11 @@ class BrowserMetrics implements Metrics {
       return
     }
 
+    if (!this.token.match(/^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/)) {
```

Comment:
Is this a check for RFC 4122 compliance? If so, this is too permissive because it accepts invalid UUIDs. `validToken` in Metrics.test.js is actually not a valid UUID, for example.

It might be better to simply use a library to validate UUIDs since implementing RFC 4122 is probably outside the scope of this project.

Reply:
Strict checking of UUID compliance is not required in this case, just that the token adheres to the string format of a UUID.

Reply:
Would that make this the corresponding validator of strings produced in uuid.ts or is the function there supposed to generate valid UUIDs?